### ISSUE TITLE
Failed to build on MSYS2

### ISSUE THREAD

**Issue by iahung** [objeck/objeck-lang#137]
Both UCRT64 and CLANG64. Please have a look @objeck @rhollines

**Comment by objeck** [comment]
I was able to build on both. What errors are you seeing?

**Comment by iahung** [comment]
> I was able to build on both. What errors are you seeing?

Did you run pacman -Syu? It used to build fine for me too. It started to have problems after pacman -Syu. You have to scroll from bottom to top to see the errors. Your build script doesn't stop if there is errors. It will just execute the next commands. You will always see it extracts api.zip regardless if the build is successful or not. The `deploy-msys2-ucrt` directory is always created but it's an incomplete distribution of Objeck since there are components failed to build. Take a closer look and you will see.

**Comment by objeck** [comment]
Not sure what is going on.

I ran ``pacman -Syu`` and it installed updates for the UCRT64 and CLANG64 shells. I was able to successfully build the toolchain in both shells without errors.

**Comment by iahung** [comment]
> Not sure what is going on.
> 
> I ran `pacman -Syu` and it installed updates for the UCRT64 and CLANG64 shells. I was able to successfully build the toolchain in both shells without errors.

This is the output of `./deploy_msys2-ucrt.sh 2> error.txt 1> output.txt` on my computer:

[error.txt](https://github.com/objeck/objeck-lang/files/10694783/error.txt)
[output.txt](https://github.com/objeck/objeck-lang/files/10694784/output.txt)

If you still have your Windows 8.1 VM, please try with an updated MSYS2 on it. I afraid it has something to do with Windows 8.1 EOL-ed since the beginning of this year, but I could be wrong.

**Comment by objeck** [comment]
Try rerunning  `panman -Syu`. I had to run the command 3-4 times before the shells fully updated.

**Comment by iahung** [comment]
> Try rerunning `pacman -Syu`. I had to run the command 3-4 times before the shells fully updated.

```
$ pacman -Syu
:: Synchronizing package databases...
 clangarm64                                                                       1078.0 KiB   201 KiB/s 00:05 [#################################################################] 100%
 mingw32                                                                          1744.4 KiB   278 KiB/s 00:06 [#################################################################] 100%
 mingw64                                                                          1804.4 KiB   272 KiB/s 00:07 [#################################################################] 100%
 ucrt64                                                                           1849.9 KiB   246 KiB/s 00:08 [#################################################################] 100%
 clang32                                                                          1722.1 KiB   148 KiB/s 00:12 [#################################################################] 100%
 clang64                                                                          1792.1 KiB   726 KiB/s 00:02 [#################################################################] 100%
 msys is up to date
:: Starting core system upgrade...
 there is nothing to do
:: Starting full system upgrade...
 there is nothing to do
```

**Comment by iahung** [comment]
> @iahung I believe this is linked to Windows 8.1, as the toolchain builds without errors on Windows 10 and 11. I do not have a Windows 8.1 VM installed.

I can't help you test building on MSYS2 anymore, I can't upgrade to newer Windows. Goodbye.

**Comment by iahung** [comment]
It's a bug on your Makefiles for UCRT64/CLANG64 @objeck @rhollines: https://github.com/msys2/MINGW-packages/issues/15587#issuecomment-1427000224

It's `/ucrt64/include` for UCRT64 and `/clang64/include` for CLANG64. But, I don't think you need to tell it to search this directory as it will be searched by default. Please recheck your Makefiles for POSIX too, remove any instances of `/usr/include`.

**Comment by objeck** [comment]
Fixed in bedd3ad

**Comment by objeck** [comment]
Fixed in bedd3ad
